TCAT/Controllers/RouteOptionsViewController.swift

```diff
@@ -327,36 +331,50 @@ class RouteOptionsViewController: UIViewController {
         routeResults.reloadData()
     }
     
-    private func getDelay(tripId: String, stopId: String) -> Future<Response<Int?>> {
-        return networking(Endpoint.getDelay(tripID: tripId, stopID: stopId)).decode()
+    private func getAllDelays(trips: [Trip]) -> Future<Response<[Delay]>> {
+        return networking(Endpoint.getAllDelays(trips: trips)).decode()
     }
-
-    @objc func updateLiveTracking(sender: Timer) {
-        // For each route in each route array inside of the 'routes' array,
-        // retrieve its delay. Use index of route to save delay for route to
-        // JSON file.
+    
+    @objc func updateAllRoutesLiveTracking(sender: Timer) {
+        var trips: [Trip] = []
+        // For each route in each route array inside of the 'routes' array, get its
+        // tripId and stopId to create trip array for request to get all delays.
         for routesArray in routes {
-            for (index, route) in routesArray.enumerated() {
-                if route.isRawWalkingRoute() { return }
-                guard let direction = route.getFirstDepartRawDirection(),
-                    let tripId = direction.tripIdentifiers?.first,
-                    let stopId = direction.stops.first?.id else {
-                        return
+            for route in routesArray {
+                if !route.isRawWalkingRoute() {
+                    guard let direction = route.getFirstDepartRawDirection(),
+                        let tripId = direction.tripIdentifiers?.first,
+                        let stopId = direction.stops.first?.id else {
+                            return
```

Comment:
Are you sure you want to `return` here and not `continue`? Return will completely stop the update. Continue will just stop processing that one route and keep trying with the next ones.

Reply:
good catch! thanks Daniel!